### PR TITLE
EHN: Create a local ramp-worker to train submission on local server using conda

### PR DESCRIPTION
**Pull request by glemaitre**

### CODE REVIEW

**Comment by glemaitre** [comment]
@jorisvandenbossche @aboucaud I am merging this PR. The tests are passing and there is a single statement not covered (when no conda environment exist which we could Mock). I would like to go further to implement the loop then.

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
     - export PATH=/home/travis/miniconda/bin:$PATH
     - conda update --yes --quiet conda
 install:
     - conda create -n ramp --yes pip python=$PYTHON_VERSION ipykernel
     - source activate ramp
-    - pip install numpy  # temporary thing
-    - pip install -r testing-requirements.txt
+    - conda install --yes numpy pyyaml sqlalchemy psycopg2
+    - conda install --yes pandas scikit-learn flask six unidecode termcolor
+    - conda install --yes pytest pytest-cov jupyter gitpython
```
[the remark this answers is not in the document]
Comment (aboucaud) [reply .travis.yml]: + 1

[the remark this answers is not in the document]
Comment (glemaitre) [reply .travis.yml]: Yep I think it makes sense. I had issue using the requirements from the different repo.
I'll do a PR.

File: ramp-engine/rampbkd/tests/test_local.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        _remove_directory(worker)
+
+
+def test_conda_worker_without_conda_env_specified(get_conda_worker):
+    worker = get_conda_worker('starting_kit')
+    # remove the conva_env parameter from the configuration
+    del worker.config['conda_env']
+    # the conda environment is set during setup; thus no need to launch
+    # submission
+    worker.setup()
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply ramp-engine/rampbkd/tests/test_local.py]: If you don't mention any `conda_env` variable in the config, the setup will use the base environment from conda. So I removed the variable from the dict and check that the setup works properly. Actually I should check the environment of the `worker._python_bin_path` and ensure that `envs` is not part of it.

**Pull request merged**